event_routing_backends/processors/xapi/event_transformers/exam_events.py

```diff
@@ -0,0 +1,97 @@
+"""
+Transformers for enrollment related events.
+"""
+
+from tincan import Activity, ActivityDefinition, Extensions, LanguageMap, Verb
+
+from event_routing_backends.processors.xapi import constants
+from event_routing_backends.processors.xapi.registry import XApiTransformersRegistry
+from event_routing_backends.processors.xapi.transformer import XApiTransformer
+
+
+class BaseExamTransformer(XApiTransformer):
+    """
+    Base transformer for exam events.
+    """
+
+    def get_object(self):
+        """
+        Get object for xAPI transformed event.
+
+        Returns:
+            `Activity`
+        """
+        object_id = self.get_data("event.exam_content_id")
+        exam_type = self.get_data("name").split(".")[2]
+
+        return Activity(
+            id=object_id,
+            definition=ActivityDefinition(
+                type=constants.XAPI_ACTIVITY_ASSESSMENT_FORMAT.format(exam_type),
```

Comment:
I think this would be more readable if we just defined each one separately like we do with other types and hard code the activity as a member variable in the subclasses.